packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -233,6 +252,29 @@ export class Secret extends SecretBase {
     return Secret.fromSecretAttributes(scope, id, { secretArn });
   }
 
+  /**
+   * Imports a secret by secret name; the ARN of the Secret will be set to the secret name.
+   * A secret with this name must exist in the same account & region.
+   */
+  public static fromSecretName(scope: Construct, id: string, secretName: string): ISecret {
+    return new class extends SecretBase {
+      public readonly encryptionKey = undefined;
+      public readonly secretArn = secretName;
```

Comment:
This should be an ARN, no?

Reply:
No. We don't have an ARN here, and intentionally do **not** want to generate something that looks like one. Valid use cases for references to a secret from other services is either the full complete arn (e.g., `arn:aws:secretsmanager....:MySecret-a3g3cf`) or the shortened secret name (e.g., `MySecret`). Creating an "incomplete" ARN (e.g., `arn:aws:secretsmanager....:MySecret`) is invalid and cannot be used anywhere. The one exception is for IAM policies, where we use the `arnForPolicies` to generate an incomplete ARN with a wildcard.